Source/Manager.swift

```diff
@@ -9,144 +9,159 @@
-        /// `/path/to/resource/123` -> `(name: "/path/to/resource", ids: [123], query: ["foo": ["bar"]])`
-        let resourceArtifacts = Manager.resourceArtifactsFromURL(request.URL!)
-        
-        /// The Resource itself
-        /// This is used to create new represetnations of a requested resource
-        let resource = Manager.resources[resourceArtifacts!.name]
-
-        /// Data that will be returned in the HTTP response.
-        /// The `name` is not returned as it is identicical to the `resourceIdentifier` on the `resource` instance
-        let dataToReturn = resource!.data(request, resourceArtifacts: (ids: resourceArtifacts!.ids, query: resourceArtifacts!.query))
-
-        let response = NSHTTPURLResponse(URL: request.URL!, statusCode: 200, HTTPVersion: "HTTP/1.1", headerFields: ["Content-Type": resource!.contentType])!
-
-        client?.URLProtocol(self, didReceiveResponse: response, cacheStoragePolicy: .NotAllowed)
-        client?.URLProtocol(self, didLoadData: dataToReturn)
-        client?.URLProtocolDidFinishLoading(self)
+  // Holds instances of Resource subclasses and are used to check if Manager can handle an incoming request
+  private static var resources = ResourceCollection<String, ResourceProtocol>()
+
+  static var count: Int {
+    return resources.count
+  }
+
+  // MARK: - Protocol implementation
+  // Class
+  override public class func canInitWithRequest(request: NSURLRequest) -> Bool {
+    // TODO: guard statement
+    let requestName = resourceArtifactsFromRequest(request)!.name
+
+    if resources[requestName] != nil {
+      return true
     }
-    
-    /// Because PopTop always returns an object it is acceptable to leave this empty.
-    override public func stopLoading() {}
-    
-    /// Returns unchanged request
-    override public class func canonicalRequestForRequest(request: NSURLRequest) -> NSURLRequest {
-        return request
+
+    return false
+  }
+
+  // MARK: - Instance
+
+  override public func startLoading() {
+    /// A tuple of (key, id, query) that will be used within the registry dict based on the URL path
+    /// `/path/to/resource/123` -> `(name: "/path/to/resource", ids: [123], query: ["foo": ["bar"]])`
+    let resourceArtifacts = Manager.resourceArtifactsFromRequest(request)
+
+    /// The Resource itself
+    /// This is used to create new represetnations of a requested resource
+    let resource = Manager.resources[resourceArtifacts!.name]
+
+    /// Data that will be returned in the HTTP response.
+    /// The `name` is not returned as it is identicical to the `resourceIdentifier` on the `resource` instance
+    let dataToReturn = resource!.data(request, resourceArtifacts: (ids: resourceArtifacts!.ids, query: resourceArtifacts!.query, body: resourceArtifacts!.body))
+
+    let response = NSHTTPURLResponse(URL: request.URL!, statusCode: 200, HTTPVersion: "HTTP/1.1", headerFields: ["Content-Type": resource!.contentType])!
```

Comment:
I'd like to make it configurable with 200 as a default.

Semi-related: I've decided to keep this project progressing at the patch level thus far. Once it we have some semblance of "done", or a more stable API, we can mark it `0.1.0`. At that point maybe we make a roadmap of what is `1.0.0`?